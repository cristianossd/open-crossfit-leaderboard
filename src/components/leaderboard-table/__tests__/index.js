import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import LeaderboardTable from '../';

describe('<LeaderboardTable />', () => {
  beforeEach(() => {
    LeaderboardTable.prototype.fetchAPI = jest.fn();
  });

  afterEach(() => {
    LeaderboardTable.prototype.fetchAPI.mockReset();
  });

  it('render correctly', () => {
    const wrapper = shallow(
      <LeaderboardTable category="isMen" filterText="" />
    );

    expect(wrapper.instance().fetchAPI).toHaveBeenCalledWith('Masculino');
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('render receive new props', () => {
    const wrapper = shallow(
      <LeaderboardTable category="isMen" filterText="" />
    );

    expect(wrapper.instance().fetchAPI).toHaveBeenCalledWith('Masculino');

    wrapper.instance().componentWillReceiveProps({
      category: 'isWomen',
      filterText: '',
    });
    expect(wrapper.instance().fetchAPI).toHaveBeenCalledWith('Feminino');
  });

  it('should filter athletes', () => {
    const wrapper = shallow(
      <LeaderboardTable category="isMen" filterText="" />
    );

    wrapper.setState({
      athletes: [
        { competitorName: 'rich froning', affiliateName: 'Mayhem' },
        { competitorName: 'matt fraser', affiliateName: 'Mayhem' },
      ],
    });

    expect(wrapper.state('athletes').length).toBe(2);

    wrapper.instance().componentWillReceiveProps({
      category: 'isMen',
      filterText: 'rich',
    });

    const filteredAthletes = wrapper.state('filteredAthletes');
    expect(filteredAthletes.length).toBe(1);
    expect(filteredAthletes[0].competitorName).toBe('rich froning');
  });
});
