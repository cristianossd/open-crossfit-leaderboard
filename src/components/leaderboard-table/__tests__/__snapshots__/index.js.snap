// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeaderboardTable /> render correctly 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-striped"
    id="leaderboardTable"
  >
    <thead
      className="thead-dark"
    >
      <tr>
        <th
          scope="col"
        >
          Atletas
        </th>
        <th
          scope="col"
        >
          18.1
        </th>
        <th
          scope="col"
        >
          18.2
        </th>
        <th
          scope="col"
        >
          18.2a
        </th>
        <th
          scope="col"
        >
          18.3
        </th>
        <th
          scope="col"
        >
          18.4
        </th>
        <th
          scope="col"
        >
          18.5
        </th>
      </tr>
    </thead>
    <tbody
      id="leaderboardBody"
    />
  </table>
  <Spinner />
</div>
`;
